rio_cogeo/cogeo.py

```diff
@@ -109,7 +109,12 @@ def cog_translate(
     config = config or {}
 
     with rasterio.Env(**config):
-        with rasterio.open(src_path) as src_dst:
+        with ExitStack() as ctx:
+            if isinstance(source, DatasetReader):
+                src_dst = ctx.enter_context(source)
```

Comment:
why do you add the `DatasetReader` to the `ExitStack`, too? Just for cleanup, so it `__exit__`s as soon as possible? Couldn't that cause trouble if the user does like [you envisioned](https://github.com/cogeotiff/rio-cogeo/issues/89#issuecomment-527911000):
```python
with MemoryFile() as memfile:
    with memfile.open(**src_profile) as mem:
         mem.write(data)
         cog_translate(mem, *args, **kwargs)
```